Source/WebKit/WebProcess/WebPage/dmabuf/AcceleratedSurfaceDMABuf.cpp

```diff
@@ -673,11 +673,13 @@ void AcceleratedSurfaceDMABuf::didRenderFrame(WebCore::Region&& damage)
         glFlush();
 
     m_target->didRenderFrame();
-    WebProcess::singleton().parentProcessConnection()->send(Messages::AcceleratedBackingStoreDMABuf::Frame(m_target->id(), WTFMove(damage), WTFMove(renderingFence)), m_id);
+    WebProcess::singleton().parentProcessConnection()->send(Messages::AcceleratedBackingStoreDMABuf::Frame(m_target->id(), m_frameDamage.region(), WTFMove(renderingFence)), m_id);
+    m_frameDamage = WebCore::Damage();
 }
 
 const WebCore::Damage& AcceleratedSurfaceDMABuf::addDamage(const WebCore::Damage& damage)
 {
+    m_frameDamage = damage;
```

Comment:
Wouldn't this be `m_frameDamage.add(damage)`? Currently if there is more than one call to `AcceleratedSurfaceDMABuf::addDamage()` only the damage passed in the last call is recorded.

```suggestion
    m_frameDamage.add(damage);
```

Reply:
I don't think so, because this is the damage for this particular frame, what we passed to didRenderFrame before. The accumulated damage is handled by the renderer already.